### ISSUE TITLE
resize() is unintuitive?

### ISSUE THREAD

**Issue by Anner-deJong** [numpy/numpy#11478]
Resizing a numpy array does NOT consider dimensions independent.
i.e. resizing an _old_  5x5x2 array into a _new_ array with `new = np.resize(old, (4, 4, 2))` simply copies the first 4*4*2 entries in _old_ into new. (e.g. `old[0, 4, :]` becomes `new[1, 0, :]`)

This is a problem if we would like simple interpolation, or if different axes represent different entities. (If the above example is an image, the top right most pixel would get warped all the way to the left, which is most likely not the intention)

I find this very unintuitive, and think there might be many more situation where some kind of interpolation, or at least a 'per-dimension-cut-off' is aimed for, than the current implemented functionality.

**Comment by njsmith** [comment]
You're right, but I'm not sure what there is to do about it. Resampling/reinterpolating an array is a complex process with many possible approaches, and too high level to include in numpy. The `resize` method is venerable and widely used, so we certainly can't change it now without breaking, like... we everything.

Would a warning in the docs have helped you avoid this confusion? Would you like to submit a PR to add such a warning?

**Comment by Anner-deJong** [comment]
Hi guys thanks for the answers.

I see, I would actually never have guessed this functionality is too high level for Numpy, I thought it was a key functionality for matrix manipulations. I ended up using `cv2.resize()`, which I guess is a similar approach as using `scipy.ndimage`. Not sure if those allow for 3< dimensions, but the main use case I can think of is 3D images anyway. 

Maybe it is an option to allow for an interpolation bool or string argument within the `np.resize()`? With the default being "" or `False` and reverting to legacy functionality, but when non empty it crosslinks to `scipy.ndimage`? Somebody with interpolation goals would probably catch that, check the docs as to which flags can be passed to the argument, and not get confused. I am unfamiliar however if and when crosslinking is acceptable coding practice.

Disregarding all that:
Not sure if a warning in the docs would have caught my attention, but it might sure do so for others. Will make a PR.
